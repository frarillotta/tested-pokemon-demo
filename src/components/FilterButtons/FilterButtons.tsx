
import React from 'react';
import styled from "styled-components";
import { PokeballButton } from '../PokeballButton/PokeballButton';
import Tooltip from '@material-ui/core/Tooltip';

type FilterButtonsProps = {
    filter: null | "captured" | "not-captured",
    setFilter: (filterType: null | "captured" | "not-captured") => void,
	className: string,
}

export const FilterButtons = ({filter, setFilter, className}: FilterButtonsProps) => {
	return (
		<Wrapper className={className}>
			<Tooltip title={"Not captured"}>
				<Pokeball
					isCaptured={false}
					isActive={filter === "not-captured"}
					onClick={() => filter === "not-captured" ? setFilter(null) : setFilter('not-captured')}
					size={45}
				/>
			</Tooltip>
			<Tooltip title={"Captured"}>
				<Pokeball
					isCaptured={true}
					isActive={filter === "captured"}
					onClick={() => filter === "captured" ? setFilter(null) : setFilter('captured')}
					size={45}
				/>
			</Tooltip>
		</Wrapper>
	)
};

const Pokeball = styled(PokeballButton)`
	height: 47px;
	border-radius: 100%;
	border: ${props => props.isActive ? "solid black 1px" : "solid white 1px"}
`;

const Wrapper = styled.main`
	display: flex;
`;
