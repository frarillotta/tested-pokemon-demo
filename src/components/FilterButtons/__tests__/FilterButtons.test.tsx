import { render } from "@testing-library/react";
import {FilterButtons} from "../FilterButtons";

describe("FilterButtons", () => {
	
	describe("when the filter is captured", () => {
		
		it("should have a consistent snapshot", () => {
			const {container} = render(<FilterButtons filter={"captured"} setFilter={jest.fn()}/>)
			expect(container).toMatchSnapshot();
		});

	});

	describe("when the filter is not-captured", () => {
		
		it("should have a consistent snapshot", () => {
			const {container} = render(<FilterButtons filter={"not-captured"} setFilter={jest.fn()}/>)
			expect(container).toMatchSnapshot();
		})
	})
	
})