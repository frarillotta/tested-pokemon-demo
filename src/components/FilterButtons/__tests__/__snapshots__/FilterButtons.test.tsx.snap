// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButtons when the filter is captured should have a consistent snapshot 1`] = `
<div>
  <main
    class="FilterButtons__Wrapper-sc-mrsqbp-1 cVlVGQ"
    id="filters"
  >
    <div
      class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph FilterButtons__Pokeball-sc-mrsqbp-0 ivNgsA"
      data-t-is-captured="false"
      title="Not captured"
    >
      <svg
        class="PokeballButton__PokeballSVG-sc-z0qx2i-1 fWWEza"
        height="45"
        viewBox="25 25 50 50"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
          fill="#f00"
          stroke="#222"
        />
        <circle
          cx="50"
          cy="50"
          fill="#222"
          r="5"
          stroke="#222"
        />
        <path
          d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
          fill="#fff"
          stroke="#222"
        />
      </svg>
    </div>
    <div
      class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph FilterButtons__Pokeball-sc-mrsqbp-0 iucRyY"
      data-t-is-captured="true"
      title="Captured"
    >
      <svg
        class="PokeballButton__PokeballSVG-sc-z0qx2i-1 iJvgOH"
        height="45"
        viewBox="25 25 50 50"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
          fill="#f00"
          stroke="#222"
        />
        <circle
          cx="50"
          cy="50"
          fill="#222"
          r="5"
          stroke="#222"
        />
        <path
          d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
          fill="#fff"
          stroke="#222"
        />
      </svg>
    </div>
  </main>
</div>
`;

exports[`FilterButtons when the filter is not-captured should have a consistent snapshot 1`] = `
<div>
  <main
    class="FilterButtons__Wrapper-sc-mrsqbp-1 cVlVGQ"
    id="filters"
  >
    <div
      class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph FilterButtons__Pokeball-sc-mrsqbp-0 iucRyY"
      data-t-is-captured="false"
      title="Not captured"
    >
      <svg
        class="PokeballButton__PokeballSVG-sc-z0qx2i-1 fWWEza"
        height="45"
        viewBox="25 25 50 50"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
          fill="#f00"
          stroke="#222"
        />
        <circle
          cx="50"
          cy="50"
          fill="#222"
          r="5"
          stroke="#222"
        />
        <path
          d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
          fill="#fff"
          stroke="#222"
        />
      </svg>
    </div>
    <div
      class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph FilterButtons__Pokeball-sc-mrsqbp-0 ivNgsA"
      data-t-is-captured="true"
      title="Captured"
    >
      <svg
        class="PokeballButton__PokeballSVG-sc-z0qx2i-1 iJvgOH"
        height="45"
        viewBox="25 25 50 50"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
          fill="#f00"
          stroke="#222"
        />
        <circle
          cx="50"
          cy="50"
          fill="#222"
          r="5"
          stroke="#222"
        />
        <path
          d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
          fill="#fff"
          stroke="#222"
        />
      </svg>
    </div>
  </main>
</div>
`;
